src/main/java/com/rtb/projectmanagementtool/project/ProjectController.java

```diff
@@ -26,60 +22,70 @@ public ProjectController(DatastoreService datastore) {
   }
 
   /**
-   * Gets all the projects in datastore that all specified users are in
+   * Gets all the projects in datastore that all users are in
    *
    * @param userIds the collection of userIds whose projects are to be retrieved
    * @return HashSet containing desired projects
    */
-  public HashSet<ProjectData> getProjects(Collection userIds) {
-
-    HashSet<ProjectData> projectContainer = new HashSet<ProjectData>();
+  public ArrayList<ProjectData> getAllProjects() {
+    return getProjects(0l, false);
+  }
 
-    PreparedQuery results = datastore.prepare(new Query("Project"));
-    for (Entity entity : results.asIterable()) {
-      if (entityContainsUser(entity, userIds)) {
-        ProjectData project = new ProjectData(entity);
-        projectContainer.add(project);
-      }
-    }
+  /**
+   * Gets all the projects in datastore that the specified user is in
+   *
+   * @param userId the id of user whose projects to retrieve
+   * @return HashSet containing desired projects
+   */
+  public ArrayList<ProjectData> getUserProjects(long userId) {
+    return getProjects(userId, false);
+  }
 
-    return projectContainer;
+  /**
+   * Gets all the projects in datastore created by user
+   *
+   * @param userId the id of owner whose projects to retrieve
+   * @return HashSet containing desired projects
+   */
+  public ArrayList<ProjectData> getCreatorProjects(long userId) {
+    return getProjects(userId, true);
   }
 
   /**
-   * Checks if the Project entity has at least one userId contained in userIds
+   * Helper function for getAllProjects(), getUserProjects(), and getCreatorProjects()
    *
-   * @param entity the project entity
-   * @param userIds the collection of userIds
-   * @return true if userIds param and entity's userIds property share at least one user
+   * @param userId the id of user whose projects to retrieve
+   * @param getCreatorProjects boolean representing the user's role in project (owner or otherwise)
+   * @return HashSet containing desired projects
    */
-  private boolean entityContainsUser(Entity entity, Collection userIds) {
-    // userIds is empty, so get all projects in database
-    if (userIds.size() == 0) {
-      return true;
-    }
+  private ArrayList<ProjectData> getProjects(long userId, boolean getCreatorProjects) {
+    ArrayList<ProjectData> projectContainer = new ArrayList<ProjectData>();
+
+    PreparedQuery results = datastore.prepare(new Query("Project"));
```

Comment:
I highly advise not pulling all projects in the data store if possible.  This would take a long time to execute and eat up a lot of memory in a real production environment.  If searching by user is a key operation for this table, I would advise reevaluating how you're storing these to be able to cleanly filter by userIDs.